lib/team.rb

```diff
@@ -103,29 +103,25 @@ def average_win_percentage(team_id)
     team_avg[team_id]
   end
 
-  ### The below block of code works iti is a total of 14 lines of logic, so we 
-  ### divide it into three seperate six logic line blocks (18 lines) and add 
-  ### all the space to it so really it's double??? Now you gotta see who calls
-  ### who and idk if having three methods or one comprehensive makes more sense.
-  ### Choose either or and lemme know. 
-
+  # Highest number of goals a particular team has scored in a single game
+  def most_goals_scored(team_id)
+    queried_team_games = @game_team_data.select do |game|
+      game[:team_id] == team_id
+    end
+    highest_scoring_game = queried_team_games.max_by do |game|
+      game[:goals]
+    end
+    highest_scoring_game[:goals].to_i
+  end
 
-  # def average_win_percentage(team_id)
-  #   total_wins = Hash.new(0)
-  #   @game_team_data.each do |row|
-  #     if row[:team_id] == team_id && row[:result] == "WIN"
-  #       total_wins[team_id] += 1
-  #     elsif row[:team_id] == team_id
-  #       total_ties_loss[team_id] += 1
-  #     end
-  #     total_ties_loss = Hash.new(0)
-  #   end
-  #   overall_hash = Hash.new(0)
-  #   total_wins.each do |team, wins|
-  #     total_games = wins + total_ties_loss[team]
-  #     percent = (wins / total_games.to_f).round(2)
-  #     overall_hash[team] = percent
-  #   end
-  #   overall_hash[team_id]
-  # end
+  # Lowest number of goals a particular team has scored in a single game
+  def fewest_goals_scored(team_id)
+    queried_team_games = @game_team_data.select do |game|
```

Comment:
This is super super cool. Love what you did with Select there. Thanks for sharing.